pyincore/semanticservice.py

```diff
@@ -0,0 +1,178 @@
+            )
+            raise
+
+    def get_all_semantic_types(
+        self,
+        hyperlink: bool = False,
+        order: str = "asc",
+        skip: int = 0,
+        limit: int = 50,
+        detail: bool = False,
+        save_json: bool = False,
+        json_path: str = None,
+        timeout: Tuple[int, int] = (30, 600),
+        **kwargs,
+    ) -> Union[list, dict]:
+        """Get all semantic types.
+
+        Args:
+            hyperlink (bool): If True, return a list of hyperlinks to the semantic types.
+            order (str): Order of the semantic types. Can be 'asc' or 'desc'.
+            skip (int): Number of semantic types to skip.
+            limit (int): Number of semantic types to return.
+            detail (bool): If True, return detailed information about the semantic types.
+            save_json (bool): If True, save the json response to a file.
+            json_path (str): Path to save the json response.
+            timeout (tuple): Timeout for the request in seconds.
+                            First value is the connection timeout, second value is the read timeout.
+            **kwargs: Arbitrary keyword arguments.
+
+        Returns:
+            list: List of semantic types.
+
+        """
+
+        url = self.base_url
+        payload = {
+            "hyperlink": hyperlink,
+            "order": order,
+            "skip": skip,
+            "limit": limit,
+            "detail": detail,
+        }
+        response = self.client.get(
+            url, params=payload, timeout=timeout, **kwargs
+        )
+        data = self.return_http_response(response).json()
+
+        if save_json:
+            # if json_path is None: then save to the default location (incore cache folder)
+            if json_path is None:
+                json_path = os.path.join(
+                    pyglobals.PYINCORE_USER_CACHE,
+                    f"semantic_types_ord_{order}_sk_{skip}_lim_{limit}.json",
+                )
+            with open(json_path, "w") as f:
+                json.dump(data, f, indent=4)
+
+        return data
+
+    def get_semantic_type_by_name(
+        self,
+        type_name: str,
+        save_json: bool = False,
+        json_path: str = None,
+        timeout: Tuple[int, int] = (30, 600),
+        **kwargs,
+    ) -> Union[list, dict]:
+        """Get semantic type by name.
+
+        Args:
+            type_name (str): Name of the semantic type.
+            save_json (bool): If True, save the json response to a file.
+            json_path (str): Path to save the json response.
+            timeout (tuple): Timeout for the request in seconds.
+                            First value is the connection timeout, second value is the read timeout.
+            **kwargs: Arbitrary keyword arguments.
+
+        Returns:
+            list or dict: List or dictionary of semantic type.
+
+        """
+
+        url = f"{self.base_url}/{type_name}"
```

Comment:
Could we use `url = urljoin(self.base_url, type_name)` instead?

Reply:
for some reason, if I did that it wasn't working as it gave not valid URL error and when I printed it out, it seemed fine

Reply:
ah maybe it's the colon? Could you look at things related to urlencoding?